aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -655,37 +655,40 @@ static void modified_bessel_k1_kernel(TensorIteratorBase& iterator) {
+          [&](char** data_, const int64_t* strides, int64_t n) {                   \
+            scalar_t* out_data = reinterpret_cast<scalar_t*>(data_[0]);            \
+            scalar_t* in_data = reinterpret_cast<scalar_t*>(data_[1]);             \
+            int64_t out_stride = strides[0] / sizeof(scalar_t);                    \
+            int64_t in_stride = strides[1] / sizeof(scalar_t);                     \
+            if (out_stride == 1 && in_stride == 1) {                               \
+              vml::v##op(out_data, in_data, n);                                    \
+              return;                                                              \
+            }                                                                      \
+            static constexpr int64_t WIDTH = (8*1024) / sizeof(scalar_t);          \
+            for (int64_t i = 0; i < n; i += WIDTH) {                               \
+              scalar_t buffer[WIDTH];                                              \
+              const int64_t width = std::min(WIDTH, n - i);                        \
+              /* If either tensor is contiguous use it, otherwise copy into */     \
+              /* a contiguous buffer so compute can still be vectorized */         \
+              scalar_t * in_buffer = in_stride == 1 ? &in_data[i] : &buffer[0];    \
+              scalar_t * out_buffer = out_stride == 1 ? &out_data[i] : &buffer[0]; \
+              if (in_stride != 1)                                                  \
+                for (const auto j : c10::irange(width))                            \
+                  in_buffer[j] = in_data[in_stride * (i + j)];                     \
+              vml::v##op(out_buffer, in_buffer, width);                            \
+              if (out_stride != 1)                                                 \
+                for (const auto j : c10::irange(width))                            \
+                    out_data[out_stride * (i + j)] = out_buffer[j];                \
+            }                                                                      \
           }
 
 #define IMPLEMENT_FLOAT_KERNEL(op)                                                  \
   inline namespace CPU_CAPABILITY {                                                 \
   void op##_kernel(TensorIteratorBase& iter) {                                      \
     TORCH_INTERNAL_ASSERT(iter.ntensors() == 2);                                    \
     AT_DISPATCH_FLOATING_TYPES_AND(kBFloat16, iter.dtype(), #op "_vml_cpu", [&]() { \
-      iter.serial_for_each(                                                         \
-          IMPLEMENT_ITERATOR_LAMBDA(op),                                            \
-          {0, iter.numel()});                                                       \
+        iter.for_each(IMPLEMENT_ITERATOR_LAMBDA(op));                               \
```

Comment:
This changes the grain size to the default one while the original code uses 2048. Do you have perf data to show it is fine? Or, maybe it is safer to keep the original grain size to avoid changes with contiguous cases?